### DISCUSSION TITLE
solved: package xxxx error 404

### DISCUSSION THREAD

**Discussion by asdf2adsfad** [Homebrew/discussions#5732] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.4-111-gd43824d
ORIGIN: https://mirrors.ustc.edu.cn/brew.git
HEAD: d43824d49f0cfb5d0bc9efa10c1c5bb0edfe4a3f
Last commit: 1 year, 9 months ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9984d2778949b441beeb63010366aefd12faa2eb
Core tap last commit: 1 year, 9 months ago
Core tap branch: master
Core tap JSON: 14 Nov 00:50 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BOTTLE_DOMAIN: https://mirrors.ustc.edu.cn/homebrew-bottles/
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 16
all_proxy: socks5://127.0.0.1:7890
http_proxy: http://127.0.0.1:7890
https_proxy: http://127.0.0.1:7890
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 12.0.5 build 1205
Git: 2.30.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1-x86_64
CLT: 12.5.0.22.9
Xcode: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Suspicious https://github.com/Homebrew/brew git origin remote found.
The current git origin is:
  https://mirrors.ustc.edu.cn/brew.git

With a non-standard origin, Homebrew won't update properly.
You can solve this by setting the origin remote:
  git -C "/usr/local/Homebrew" remote set-url origin https://github.com/Homebrew/brew

Warning: No Cask quarantine support available: unknown reason.

Warning: Suspicious https://github.com/Homebrew/homebrew-core git origin remote found.
The current git origin is:
  https://mirrors.ustc.edu.cn/homebrew-core.git

With a non-standard origin, Homebrew won't update properly.
You can solve this by setting the origin remote:
  git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" remote set-url origin https://github.com/Homebrew/homebrew-core

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1

Warning: gettext files detected at a system prefix.
These files can cause compilation and link failures, especially if they
are compiled with improper architectures. Consider removing these files:
  /usr/local/lib/libgettextlib.dylib
  /usr/local/lib/libintl.dylib
  /usr/local/include/libintl.h
  /usr/local/lib/libgettextlib.dylib
  /usr/local/lib/libintl.dylib
  /usr/local/include/libintl.h
Error: unknown or unsupported macOS version: :dunno
```

### Description of issue

brew install autossh

Running `brew update --auto-update`...
Warning: You are using macOS 15.
We do not provide support for this pre-release version.
It is expected behaviour that some formulae will fail to build in this pre-release version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.

==> Fetching autossh
**==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/f6fcb8f
curl: (56) The requested URL returned error: 404**                               

Error: autossh: Failed to download resource "autossh--ruby-source"
Download failed: https://raw.githubusercontent.com/Homebrew/homebrew-core/f6fcb8f9252ec88e26bea12b4c68321a8266a2f5/Formula/autossh.rb
==> `brew cleanup` has not been run in the last 30 days, running now...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
Error: Permission denied - /usr/local/share/ghostscript/9.53.3/Resource/Font/bchb8a.pfb

**Comment by gromgit** [comment, marked as answer]
>```
> HOMEBREW_VERSION: 4.0.4-111-gd43824d
> ORIGIN: https://mirrors.ustc.edu.cn/brew.git
> HEAD: d43824d49f0cfb5d0bc9efa10c1c5bb0edfe4a3f
> Last commit: 1 year, 9 months ago
>```

Your Homebrew software is almost 2 years old, and that will cause problems beyond what you're experiencing now.

If you haven't run `brew update` recently, do it right now. If you _have_ been running `brew update` regularly, this means your current git remote is VERY out of date, so either switch back to the original git remote:
>```
>Warning: Suspicious https://github.com/Homebrew/brew git origin remote found.
>The current git origin is:
>   https://mirrors.ustc.edu.cn/brew.git
>
> With a non-standard origin, Homebrew won't update properly.
> You can solve this by setting the origin remote:
>   git -C "/usr/local/Homebrew" remote set-url origin https://github.com/Homebrew/brew
>```

or ask the mirror maintainers at ustc.edu.cn to update their repository.
